Run black on python generated code

joeldodge79 opened the pull request:
Format the generated code as cleanly as possible using black
- Uses pipenv to install/update black and run it
- slight refactor on utils.run - no need for async and custom error

jkaster commented on utils.ts:
```diff
@@ -61,7 +61,7 @@ export const fail = (name: string, message: string) => {
 }
 
-export const run = async (command: string, args: string[]) => {
+export const run = (command: string, args: string[], errMsg?: string) => {
```
Nice refactor. I originally had a `run` that was async, but never got back to changing this. I like the error message being passed in also.

joeldodge79 replied:
yeah, I saw that so I figured this was safe

joeldodge79 commented:
Here's some sample output:

1/ no `pipenv` installed errors out telling me to install it
```bash
› yarn sdk
yarn run v1.17.3
$ ts-node sdkGen.ts
generating python ...
319 API methods
211 API models: 160 Spec, 0 Request, 51 Write
Error: Error, please install pipenv: https://docs.pipenv.org/en/latest/install/#installing-pipenv
Error: please install pipenv: https://docs.pipenv.org/en/latest/install/#installing-pipenv
    at Object.exports.quit (/Users/joeldodge/Documents/sdk_codegen/utils.ts:46:13)
    at Object.exports.run (/Users/joeldodge/Documents/sdk_codegen/utils.ts:78:12)
    at PythonFormatter.reformatFile (/Users/joeldodge/Documents/sdk_codegen/python.fmt.ts:350:7)
    at PythonFormatter.reformat (/Users/joeldodge/Documents/sdk_codegen/codeFormatter.ts:239:27)
    at Object.<anonymous> (/Users/joeldodge/Documents/sdk_codegen/sdkGen.ts:67:39)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/joeldodge/Documents/sdk_codegen/sdkGen.ts:28:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:832:11)
    at Object.<anonymous> (/Users/joeldodge/Documents/sdk_codegen/node_modules/ts-node/src/bin.ts:154:12)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

2/ now I have `pipenv` installed, first run takes a little while since it has to create the virtualenv:

```bash
› yarn sdk
yarn run v1.17.3
$ ts-node sdkGen.ts
generating python ...
319 API methods
211 API models: 160 Spec, 0 Request, 51 Write
Creating a virtualenv for this project…
Pipfile: /Users/joeldodge/Documents/sdk_codegen/Pipfile
Using /Users/joeldodge/.pyenv/versions/3.7.4/bin/python3 (3.7.4) to create virtualenv…
⠧ Creating virtual environment...Using base prefix '/Users/joeldodge/.pyenv/versions/3.7.4'
New python executable in /Users/joeldodge/.local/share/virtualenvs/sdk_codegen-x22ukLCf/bin/python3
Also creating executable in /Users/joeldodge/.local/share/virtualenvs/sdk_codegen-x22ukLCf/bin/python
Installing setuptools, pip, wheel...
done.
Running virtualenv with interpreter /Users/joeldodge/.pyenv/versions/3.7.4/bin/python3

✔ Successfully created virtual environment!
Virtualenv location: /Users/joeldodge/.local/share/virtualenvs/sdk_codegen-x22ukLCf
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
⠹ Locking..✔ Success!
Updated Pipfile.lock (595abd)!
reformatted /Users/joeldodge/Documents/sdk_codegen/python/looker_sdk/sdk/methods.py
reformatted /Users/joeldodge/Documents/sdk_codegen/python/looker_sdk/sdk/models.py
All done! ✨ 🍰 ✨
2 files reformatted, 1 file left unchanged.
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
⠙ Locking..✔ Success!
Updated Pipfile.lock (595abd)!
All done! ✨ 🍰 ✨
3 files left unchanged.
generating typescript ...
319 API methods
248 API models: 160 Spec, 37 Request, 51 Write
reformatted ./typescript/looker/sdk/methods.ts,./typescript/looker/sdk/models.ts
✨  Done in 16.69s.
```

3/ subsequent runs are faster (virtualenv already exists):

```bash
› yarn sdk
yarn run v1.17.3
$ ts-node sdkGen.ts
generating python ...
319 API methods
211 API models: 160 Spec, 0 Request, 51 Write
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
⠹ Locking..✔ Success!
Updated Pipfile.lock (595abd)!
reformatted /Users/joeldodge/Documents/sdk_codegen/python/looker_sdk/sdk/methods.py
reformatted /Users/joeldodge/Documents/sdk_codegen/python/looker_sdk/sdk/models.py
All done! ✨ 🍰 ✨
2 files reformatted, 1 file left unchanged.
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
⠹ Locking..✔ Success!
Updated Pipfile.lock (595abd)!
All done! ✨ 🍰 ✨
3 files left unchanged.
generating typescript ...
319 API methods
248 API models: 160 Spec, 37 Request, 51 Write
reformatted ./typescript/looker/sdk/methods.ts,./typescript/looker/sdk/models.ts
✨  Done in 11.88s.
```
